source/AI.cpp

```diff
@@ -824,11 +825,19 @@ bool AI::FollowOrders(Ship &ship, Command &command) const
 	}
 	
 	shared_ptr<Ship> target = it->second.target.lock();
-	if(type == Orders::MOVE_TO && ship.Position().Distance(it->second.point) > 20.)
-		MoveTo(ship, command, it->second.point, Point(), 10., .1);
+	if(type == Orders::MOVE_TO && it->second.targetSystem && ship.GetSystem() != it->second.targetSystem)
+	{
+		// The desired position is in a different system.
+		DistanceMap distance(ship, it->second.targetSystem);
+		const System *to = distance.Route(ship.GetSystem());
+		ship.SetTargetSystem(to);
+		return false;
+	}
 	else if(type == Orders::HOLD_POSITION || type == Orders::MOVE_TO)
 	{
-		if(ship.Velocity().Length() > .001 || !ship.GetTargetShip())
+		if(it->second.point && ship.Position().Distance(it->second.point) > 20.)
+			MoveTo(ship, command, it->second.point, Point(), 10., .1);
```

Comment:
Rather than moving these these two lines into this if clause, it would be clearer to just turn the original "if" in line 827 into an "else if."

That would limit the code changes to what's strictly necessary (just adding yout one new "if" clause for moving to a different system). Also, checking `type == Orders::MOVE_TO` is clearer than checking `it->second.point`.